packages/build-tools/src/utils/project.ts

```diff
@@ -22,6 +23,15 @@ export async function setup<TJob extends Job>(ctx: BuildContext<TJob>): Promise<
     }
   });
 
+  // this must be the first build phase to make sure package.json exists and is valid
```

Comment:
Ideally, I would prefer if package.json was printed just before app config, you could add validation in PREPARE_PROJECT phase and keep printing package.json in the original place. No strong opinion on that so I'm leaving it up to you.